tests/test_representation.py

```diff
@@ -21,131 +23,215 @@ def load_tests(loader, tests, ignore):
+    ):
+        if int_or_float == "int":
+            s_true = pd.Series(
+                correct_output_values,
+                index=s_tokenized_output_noncontinuous_index,
+                dtype="int",
+            ).astype(pd.SparseDtype(np.int64, 0))
+        else:
+            s_true = pd.Series(
+                correct_output_values,
+                index=s_tokenized_output_noncontinuous_index,
+                dtype="float",
+            ).astype(pd.SparseDtype("float", np.nan))
+
+        result_s = test_function(s_tokenized_with_noncontinuous_index)
+
+        pd.testing.assert_series_equal(s_true, result_s)
+
+    @parameterized.expand(test_cases_vectorization_min_df)
+    def test_vectorization_min_df(
+        self, name, test_function, correct_output_values, int_or_float
+    ):
+        if int_or_float == "int":
+            s_true = pd.Series(
+                correct_output_values,
+                index=s_tokenized_output_min_df_index,
+                dtype="int",
+            ).astype(pd.SparseDtype(np.int64, 0))
+        else:
+            s_true = pd.Series(
+                correct_output_values,
+                index=s_tokenized_output_min_df_index,
+                dtype="float",
+            ).astype(pd.SparseDtype("float", np.nan))
+
+        result_s = test_function(s_tokenized, min_df=2)
+
+        pd.testing.assert_series_equal(s_true, result_s)
+
+    @parameterized.expand(test_cases_vectorization)
+    def test_vectorization_not_tokenized_yet_warning(self, name, test_function, *args):
         with self.assertWarns(DeprecationWarning):  # check raise warning
-            representation.count(s)
+            test_function(s_not_tokenized)
+
+    @parameterized.expand(test_cases_vectorization)
+    def test_vectorization_arguments_to_sklearn(self, name, test_function, *args):
+        try:
+            test_function(s_not_tokenized, max_features=1, min_df=1, max_df=1.0)
+        except TypeError:
+            self.fail("Sklearn arguments not handled correctly.")
 
     """
-    TF-IDF
+    Individual / special tests.
     """
 
     def test_tfidf_formula(self):
         s = pd.Series(["Hi Bye", "Test Bye Bye"])
         s = preprocessing.tokenize(s)
+        s_true_index = pd.MultiIndex.from_tuples(
+            [(0, "Bye"), (0, "Hi"), (1, "Bye"), (1, "Test")], names=["document", "word"]
+        )
         s_true = pd.Series(
             [
-                [
-                    1.0 * (math.log(3 / 3) + 1),
-                    1.0 * (math.log(3 / 2) + 1),
-                    0.0 * (math.log(3 / 2) + 1),
-                ],
-                [
-                    2.0 * (math.log(3 / 3) + 1),
-                    0.0 * (math.log(3 / 2) + 1),
-                    1.0 * (math.log(3 / 2) + 1),
-                ],
-            ]
-        )
-        s_true.rename_axis("document", inplace=True)
-        self.assertEqual(representation.tfidf(s), s_true)
+                1.0 * (math.log(3 / 3) + 1),
```

Comment:
We might again here use the same `_tfidf` function used before.

Reply:
Will do